pulser/simresults.py

```diff
@@ -125,7 +125,7 @@ def sample_final_state(self, meas_basis=None, N_samples=1000):
 
         N = self._size
         self.N_samples = N_samples
-        probs = np.abs(self._states[-1])**2
+        probs = np.abs(self._states[-1].data.toarray())**2
```

Comment:
I'd suggest using `Qobj.full()` to access the ndarray, to comply with QuTiP's preferred method.

Reply:
Damn, I just saw the approval and didn't notice this suggestion before merging. I'll sneakily add this in when I bump the version.